merlin/datasets/ecommerce/booking/dataset.py

```diff
@@ -0,0 +1,237 @@
+):
+    train = get_lib().read_csv(path / "train_set.csv")
+    test = get_lib().read_csv(path / "test_set.csv")
+
+    train["checkin"] = get_lib().to_datetime(train["checkin"], format="%Y-%m-%d")
+    train["checkout"] = get_lib().to_datetime(train["checkout"], format="%Y-%m-%d")
+    test["checkin"] = get_lib().to_datetime(test["checkin"], format="%Y-%m-%d")
+    test["checkout"] = get_lib().to_datetime(test["checkout"], format="%Y-%m-%d")
+
+    (path / "train").mkdir(exist_ok=True)
+    (path / "test").mkdir(exist_ok=True)
+
+    train.to_parquet(path / "train/data.parquet")
+    test.to_parquet(path / "test/data.parquet")
+
+
+def transform_booking(
+    data: Union[str, Path, Tuple[merlin.io.Dataset, merlin.io.Dataset]],
+    output_path: Union[str, Path],
+    nvt_workflow=None,
+    **kwargs,
+):
+    """_summary_
+
+    Args:
+        data (Union[str, Path, Tuple[merlin.io.Dataset, merlin.io.Dataset]]): _description_
+        output_path (Union[str, Path]): _description_
+        nvt_workflow (_type_, optional): _description_. Defaults to None.
+
+    Raises:
+        ValueError: _description_
+    """
+    nvt_workflow = nvt_workflow or default_booking_transformation(**locals())
+
+    if isinstance(data, (str, Path)):
+        _train = glob(str(data / "train/*.parquet"))
+        _valid = glob(str(data / "test/*.parquet"))
+    elif (
+        isinstance(data, tuple)
+        and len(data) == 2
+        and all(isinstance(x, merlin.io.Dataset) for x in data)
+    ):
+        _train, _valid = data
+    else:
+        raise ValueError("data must be a path or a tuple of train and valid datasets")
+
+    workflow_fit_transform(nvt_workflow, _train, _valid, str(output_path), **kwargs)
+
+
+def default_booking_transformation(**kwargs):
+    df_season = get_lib().DataFrame(
+        {"month": range(1, 13), "season": ([0] * 3) + ([1] * 3) + ([2] * 3) + ([3] * 3)}
+    )
+
+    month = (
+        ["checkin"]
+        >> nvt.ops.LambdaOp(lambda col: col.dt.month.astype("int64"))
+        >> nvt.ops.Rename(name="month")
+    )
+
+    weekday_checkin = (
+        ["checkin"]
+        >> nvt.ops.LambdaOp(lambda col: col.dt.weekday)
+        >> nvt.ops.Rename(name="weekday_checkin")
+    )
+
+    weekday_checkout = (
+        ["checkout"]
+        >> nvt.ops.LambdaOp(lambda col: col.dt.weekday)
+        >> nvt.ops.Rename(name="weekday_checkout")
+    )
+
+    is_weekend = (
+        weekday_checkin
+        >> nvt.ops.LambdaOp(lambda col: col.isin([5, 6]).astype(int))
+        >> nvt.ops.Rename(name="is_weekend")
+    )
+
+    length_feature = (
+        ["checkout"]
+        >> nvt.ops.LambdaOp(lambda col, df: (col - df["checkin"]).dt.days, dependency=["checkin"])
+        >> nvt.ops.Rename(name="length")
+    )
+
+    season = month >> nvt.ops.JoinExternal(
+        df_season,
+        on="month",
+        how="left",
+    )
+
+    cityid = (
+        ["city_id"] >> nvt.ops.Categorify(freq_threshold=9) >> nvt.ops.AddTags(tags=[Tags.ITEM_ID])
```

Comment:
I used start index = 1 arg in categorify bcs we dont want the OOV's to be `0` and padded to be `0` at the same time.,since I used `start_index` in categorify, Idk  how frequency_threshold really works with `start_index`.

also `frequence_threshold=9`  a big value. 30% of the cities were visited only once and 50% visited only 3s. so if we apply frequency threshold we combine many of them to one unique embedding value. I'd remove this.